allennlp/nn/initializers.py

```diff
@@ -193,9 +214,22 @@ def __init__(self, initializers: List[Tuple[str, Initializer]] = None) -> None:
         initializers : ``List[Tuple[str, Initializer]]``, optional (default = [])
             A list mapping parameter regexes to initializers.  We will check each parameter against
             each regex in turn, and apply the initializer paired with the first matching regex, if
-            any.
+            any. If "prevent" is assigned to any regex, then it will override and prevent the matched
+            parameters to be initialzed.
         """
-        self._initializers = initializers or []
+
+        self._initializers: List[Tuple[str, Initializer]] = []
+        initializers = initializers or []
+        self._prevent_regex = None
+        prevent_regexes = []
+        for initializer in initializers:
+            if isinstance(initializer[1], PreventRegexInitializer):
+                prevent_regexes.append(initializer[0])
+            else:
+                self._initializers.append(initializer)
+        if prevent_regexes:
+            self._prevent_regex = "("+")|(".join(prevent_regexes)+")"
```

Comment:
Spaces around operators would make this a lot easier to read.

Reply:
True. I will fix this. Thanks